Update scRNA-seq chapter and add a paragraph on quantification of gene expression

LuisHeinzlmeier opened the pull request:
- shortened the first paragraphs (based on ideas from https://github.com/theislab/single-cell-best-practices/pull/329#discussion_r1957203265).
- added dropdowns for benefits and limitations (using icons [1](https://fontawesome.com/icons/thumbs-up?f=classic&s=solid) and [2](https://fontawesome.com/icons/thumbs-down?f=classic&s=solid))
- added a table for the comparison of sequencing technologies across generations
- improved the structure and readability of the `single-cell RNA sequencing`- paragraph
- added recent news/literature
- added `quantifying gene expression`-paragraph with a new figure
- added self-assessment questions
- added `mRNA` and `sequencing` to the glossary and linked the terms in most chapters (at least in those we have already worked through)
- Closes #368
- Closes #328

Zethson commented on jupyter-book/introduction/scrna_seq.md:
```diff
@@ -11,41 +11,29 @@ Most definitions of the term life share a common entity - {term}`cells <Cell>`.
 _Sequencing_ is the process of deciphering the order of DNA nucleotides.
 It is primarily used to unveil the genetic information that is carried by a specific DNA segment, a complete genome, or even a complex microbiome.
-DNA _sequencing_ allows researchers to identify the location and function of genes and regulatory elements in the DNA molecule and the genome and uncovers genetic features such as open reading frames (ORFs) or {term}`CpG` islands, which indicate promotor regions.
+DNA _sequencing_ allows researchers to identify a gene's location, function, and regulation. For example, it uncovers genetic features such as open reading frames (ORFs) or {term}`CpG` islands, which indicate promotor regions.
```
Not sure whether we need DNA sequencing and above sequencing to be italic. We may as well just have a glossary item.

Zethson replied:
ORFs can be a glossary item.

LuisHeinzlmeier replied:
I removed the italics and added sequencing to the glossary.

Zethson commented on jupyter-book/introduction/scrna_seq.md:
```diff
@@ -166,44 +179,85 @@ The differences lie primarily in the chemistry of the respective sequencing tech
+While the number of genes can vary slightly between individuals (~70 genes), the average human genome contains roughly 22,000 genes {cite}`pertea2010between`.
+Gene transcription is far from continuous.
+Instead, it occurs in stochastic bursts — short, irregular periods of activity during which a gene might suddenly produce multiple mRNA transcripts before returning to silence {cite}`suter2011mammalian`.
+This is also the reason why we model mRNA transcription with a negative binomial distribution.
```
The NBB might need 0.5 sentences of explanation here. Not where it's coming from, because you are nicely doing that, but its properties

LuisHeinzlmeier replied:
Added two useful properties from negative binomial distribution.

Zethson commented on jupyter-book/introduction/scrna_seq.md:
```diff
@@ -239,37 +293,56 @@ This is especially true for sequences that are amplified at a later stage of the
+The most widely used protocols are **inDrop** {cite}`Klein2015`, **Drop-seq** {cite}`exp:Macosko2015` and the commercially available **10x Genomics Chromium** {cite}`exp:Zheng2017`.
+These protocols use microfluids to trap cells in tiny, watery bubbles known as droplets.
+Each droplet forms a separate spaces containing only one cell and the required chemicals (beads).
+The mentioned protocols are able to generate droplets several thousand times per second.
```
```suggestion
The mentioned protocols can to generate droplets thousands of times per second.
```

Nit: More idiomatic

LuisHeinzlmeier replied:
Adapted the sentence!

Zethson reviewed (commented):
This is much better and I really like what you're doing here. This is great work already!

1. It's a bit dangerous to talk so much about the central dogma because everyone knows that it's limited and does not paint the whole picture. A biologist will notice this while reading :)
2. Nit: You missed a few new lines for sentences.
3. I know that it's annoying work but I think we can link to a few more existing glossary items in a few places to make some things clearer. I trust your judgement.

Zethson commented:
If necessary, yes. @seohyonkim ?

LuisHeinzlmeier commented:
Thank you very much for the feedback!

1. Avoided using the term central dogma that often. Also, read [this short article](https://pmc.ncbi.nlm.nih.gov/articles/PMC7649963/) based on your comment. Thank you for drawing attention to this 💪 

2. Added a couple of new lines.

3. Added `mRNA` and `sequencing` to the glossary and linked the terms in most chapters (at least in those we have already worked through).

Zethson reviewed (approved):
Thank you!

> More details on this computational process will be described in the next chapter.

Ideally that would be a sphinx link.

LuisHeinzlmeier commented:
> Thank you!
> 
> > More details on this computational process will be described in the next chapter.
> 
> Ideally that would be a sphinx link.

Added a link to the next chapter.
I think then it is ready to be merged from my side! 🚀
